ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/JavaCodeGenerator.java

```diff
@@ -84,15 +80,66 @@ public void generate(
     }
 
     private void generateCommands() throws IOException, TemplateException {
+        generateAllRequests();
+        generateAllModels();
+        generateClient();
+    }
+
+    private void generateAllModels() throws IOException, TemplateException {
+        if (spec.getTypes() == null || spec.getTypes().isEmpty()) {
+            return;
```

Comment:
We may want to log that there was nothing for us to generate.  It could help us spot an error when debugging.